core/src/main/java/apoc/util/s3/S3ParamsExtractor.java

```diff
@@ -45,17 +52,17 @@ public static S3Params extract(URL url) throws  IllegalArgumentException {
         }
 
         // endpoint
-        String endpoint = url.getHost();
+        String endpoint = uri.getHost();
 
```

Comment:
There's a problem with this (and that's why the `LoadS3Test` tests are failing). The example I give is with http addresses, but it would be similar for S3 ones:

```scala
import java.net.{URL, URI}

new URL("http://us-east-1.127.0.0.1:55220/test-bucket/test.csv?accessKey=accesskey&secretKey=secretkey").getHost 
// us-east-1.127.0.0.1

new URI("http://us-east-1.127.0.0.1:55220/test-bucket/test.csv?accessKey=accesskey&secretKey=secretkey").getHost
// null
```

Reply:
I experimented with a fix for that in 803800d. Feel free to discard it if you have a better suggestion